### PR TITLE
Lab№4 Федосеев Валера 381906-2

### PR DESCRIPTION
**Pull request by Fedoseevv26**

### CODE REVIEW

**Comment by Fedoseevv26** [comment]
@ggladilov, please review

File: llvm/lib/Transforms/PeepholeOptimizationCourse/ObfuscateI8AddNewPassManager/ObfuscateI8AddNewPassManager.cpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            auto And = builder.CreateAnd(a,b);
+            auto Xor = builder.CreateXor(a,b);
+            auto MulW2 = builder.CreateMul(Xor,Const2);
+            auto Add = builder.CreateAdd(And,MulW2);
+            auto MulW39 = builder.CreateMul(Add,Const39);
+            auto AddW23 = builder.CreateAdd(MulW39,Const23)
+            auto MulW151 = builder.CreateMul(Sum,Const151);
+
+            auto obfuscated = BinaryOperator::CreateAdd(MulW151,Const111);
+            ReplaceInstWithInst(basicBlock->getInstList(), instruction, obfuscated);
```
Comment (ggladilov) [inline llvm/lib/Transforms/PeepholeOptimizationCourse/ObfuscateI8AddNewPassManager/ObfuscateI8AddNewPassManager.cpp]: Could you please reuse that code between passes?

File: llvm/lib/Transforms/PeepholeOptimizationCourse/ObfuscateI8AddFunction/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+add_llvm_component_library(
+	LLVMObfuscateI8AddFunction
```
Comment (ggladilov) [inline llvm/lib/Transforms/PeepholeOptimizationCourse/ObfuscateI8AddFunction/CMakeLists.txt:2]: I don't see where you link against that library. How did linker find implentations?

File: llvm/lib/Transforms/PeepholeOptimizationCourse/ObfuscateI8AddNewPassManager/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+add_llvm_component_library(
+  LLVMObfuscateI8AddNewPassManager
```
Comment (ggladilov) [inline llvm/lib/Transforms/PeepholeOptimizationCourse/ObfuscateI8AddNewPassManager/CMakeLists.txt:2]: I don't see where you link against that library. How did linker find implentations?

**Comment by ggladilov** [comment]
Please re-check update task definition [here](https://github.com/ggladilov/llvm-project/blob/a3de078582c71ced3e93d87ec507506071ddb969/llvm/lib/Transforms/PeepholeOptimizationCourse/Readme.md).

File: llvm/lib/Transforms/PeepholeOptimizationCourse/ObfuscateI8AddNewPassManager/CMakeLists.txt (hunk 3 above)
Comment (Fedoseevv26) [reply llvm/lib/Transforms/PeepholeOptimizationCourse/ObfuscateI8AddNewPassManager/CMakeLists.txt:2]: fix it here [f93f764](https://github.com/ggladilov/llvm-project/pull/6/commits/f93f7642514533f5c74f78c3e2af6e082cccd8e4)

File: llvm/lib/Transforms/PeepholeOptimizationCourse/ObfuscateI8AddFunction/CMakeLists.txt (hunk 2 above)
Comment (Fedoseevv26) [reply llvm/lib/Transforms/PeepholeOptimizationCourse/ObfuscateI8AddFunction/CMakeLists.txt:2]: fix it here [f93f764](https://github.com/ggladilov/llvm-project/pull/6/commits/f93f7642514533f5c74f78c3e2af6e082cccd8e4)

**Pull request closed without merging**